Storage/SQL/SQLiteHistory.swift

```diff
@@ -124,12 +124,16 @@ private let topSitesQuery = "SELECT * FROM \(TableCachedTopSites) ORDER BY frece
 extension SQLiteHistory: BrowserHistory {
     public func removeSiteFromTopSites(site: Site) -> Success {
         if let host = site.url.asURL?.normalizedHost() {
-            return db.run([("UPDATE \(TableDomains) set showOnTopSites = 0 WHERE domain = ?", [host])])
-                >>> { return self.refreshTopSitesCache() }
+            return self.removeHostFromTopSites(host)
         }
         return deferMaybe(DatabaseError(description: "Invalid url for site \(site.url)"))
     }
 
+    public func removeHostFromTopSites(host: String) -> Success {
```

Comment:
So this got me thinking. We don't use the existing `removeSiteFromTopSites` method because we're using `TopSiteItem` instead of `Site` in our top sites source. Looking back at where we convert a `Site` to `TopSiteItem` [1] it kind of seems pointless. Why not just use `Site` all the way through? Would also make it more flexible in case we don't want to show a top site but other kinds of sites as well.

[1] https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Home/ActivityStreamPanel.swift#L271